### ISSUE TITLE
Exception during loading effects for the ledger:	{"Unknown 'type_i'='26'"}

### ISSUE THREAD

**Issue by OttoVT** [elucidsoft/dotnet-stellar-sdk#313]
**Describe the bug**
Exception during loading effects for the ledger 384048:	{"Unknown 'type_i'='26'"}

https://laboratory.stellar.org/#explorer?resource=effects&endpoint=for_ledger&values=eyJsZWRnZXIiOiIzODIzNDkifQ%3D%3D&network=test

**To Reproduce**
```
var server = new Server("https://horizon-testnet.stellar.org");

//Exception is thrown here!
var page = await _server.Effects.ForLedger(384048)
                .Limit(100)
                .Execute();
```

**Expected behavior**
There should be no exception!

**Screenshots**
![image](https://user-images.githubusercontent.com/14955144/114367694-bf78ef80-9b8d-11eb-8dc9-0431f150f73e.png)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: PC
 - OS: Microsoft Windows 10 Pro 
 - Version 10.0.19041 Build 19041

**Additional context**

$exception | {"Unknown 'type_i'='26'"} | Newtonsoft.Json.JsonSerializationException

Stack Trace:
   at stellar_dotnet_sdk.responses.EffectDeserializer.CreateResponse(Int32 type)
   at stellar_dotnet_sdk.responses.EffectDeserializer.ReadJson(JsonReader reader, Type objectType, EffectResponse existingValue, Boolean hasExistingValue, JsonSerializer serializer)
   at Newtonsoft.Json.JsonConverter`1.ReadJson(JsonReader reader, Type objectType, Object existingValue, JsonSerializer serializer)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.DeserializeConvertable(JsonConverter converter, JsonReader reader, Type objectType, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateList(IList list, JsonReader reader, JsonArrayContract contract, JsonProperty containerProperty, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonConverter[] converters)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonConverter[] converters)
   at stellar_dotnet_sdk.responses.JsonSingleton.GetInstance[T](String content)
   at stellar_dotnet_sdk.requests.ResponseHandler`1.<HandleResponse>d__0.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at stellar_dotnet_sdk.requests.RequestBuilder`1.<Execute>d__7`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at stellar_dotnet_sdk.requests.RequestBuilderExecutePageable`2.<Execute>d__1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()

**Comment by Kirbyrawr** [comment]
Hi, will check it soon!

**Comment by Kirbyrawr** [comment]
Hi, type_id 26 is a new effect response in protocol 16, i don't see where that 26 is coming from since in the JSON data you shared (the url) it doesn't appear.

**Comment by Kirbyrawr** [comment]
What branch are you using, master?

**Comment by OttoVT** [comment]
<PackageReference Include="stellar-dotnet-sdk" Version="7.0.12" />

**Comment by OttoVT** [comment]
"stellar-dotnet-sdk" Version="7.0.12"

**Comment by OttoVT** [comment]
You may find type 26 in 384048 now -_-

**Comment by Kirbyrawr** [comment]
We have an ongoing PR #312 that fix this problem

**Comment by KonstantinRyazantsev** [comment]
@Kirbyrawr is there any estimation on when this fix will be released? Is it possible to support new protocol features in advance, before the mainnet is upgraded?

We use dotnet-stellar-sdk lib in our production services and issues like this affect us significantly. 

Can we help to avoid this somehow?

**Comment by Kirbyrawr** [comment]
You are correct, @elucidsoft and @fracek can tell you more about this.

Usually Stellar Foundation tell us to update our SDK's before it goes to mainnet, there was some issues with validators and that protocol update didn't include rhis changes, but right now the PR is done we only need to merge it and of course it must be done before its in mainnet.

**Comment by elucidsoft** [comment]
We have just released v7.1.0 which should fix this issue.

**Comment by OttoVT** [comment]
thx
